### ISSUE TITLE
Execution failed for task ':games_services:compileDebugKotlin'.

### ISSUE THREAD

**Issue by robertnicjoo** [Abedalkareem/games_services#159]
I'm not able to run my app in android emulator, it returns following error:

```bash
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':games_services:compileDebugKotlin'.
> Inconsistent JVM-target compatibility detected for tasks 'compileDebugJavaWithJavac' (1.8) and 'compileDebugKotlin' (17).
```

Here is my files settings:

`android/build.gradle`

```gradle
buildscript {
    ext.kotlin_version = '1.9.23'
    repositories {
        google()
        mavenCentral()
    }

    dependencies {
        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
        classpath 'com.google.gms:google-services:4.4.1'
        classpath 'com.google.firebase:firebase-crashlytics-gradle:2.9.9'
    }
}


allprojects {
    repositories {
        google()
        mavenCentral()
    }
}

rootProject.buildDir = '../build'
subprojects {
    project.buildDir = "${rootProject.buildDir}/${project.name}"
}
subprojects {
    project.evaluationDependsOn(':app')
}

tasks.register("clean", Delete) {
    delete rootProject.buildDir
}
```

`android/app/build.gradle`

```gradle
plugins {
    id "com.android.application"
    id "kotlin-android"
    id "dev.flutter.flutter-gradle-plugin"
    id 'com.google.gms.google-services'
}

//....

compileOptions {
  sourceCompatibility JavaVersion.VERSION_1_8
  targetCompatibility JavaVersion.VERSION_1_8
}

kotlinOptions {
  jvmTarget = '1.8'
}
//...
```

`settings.gradle`

```gradle
pluginManagement {
    def flutterSdkPath = {
        def properties = new Properties()
        file("local.properties").withInputStream { properties.load(it) }
        def flutterSdkPath = properties.getProperty("flutter.sdk")
        assert flutterSdkPath != null, "flutter.sdk not set in local.properties"
        return flutterSdkPath
    }
    settings.ext.flutterSdkPath = flutterSdkPath()

    includeBuild("${settings.ext.flutterSdkPath}/packages/flutter_tools/gradle")

    repositories {
        google()
        mavenCentral()
        gradlePluginPortal()
    }

    plugins {
        id "dev.flutter.flutter-gradle-plugin" version "1.0.0" apply false
    }
}

plugins {
    id "dev.flutter.flutter-plugin-loader" version "1.0.0"
    id "com.android.application" version '8.3.1' apply false
}

include ":app"
```

`gradle.properties`

```gradle
org.gradle.jvmargs=-Xmx4G
android.useAndroidX=true
android.enableJetifier=true
```

`gradle-wrapper.properties`

```gradle
distributionBase=GRADLE_USER_HOME
distributionPath=wrapper/dists
zipStoreBase=GRADLE_USER_HOME
zipStorePath=wrapper/dists
distributionUrl=https\://services.gradle.org/distributions/gradle-8.4-bin.zip
```

**Comment by theLee3** [comment]
Try updating the following in `android/app/build.gradle`:
```
kotlinOptions {
    jvmTarget = JavaVersion.VERSION_1_8
}
```

**Comment by manjav** [comment]
It seems new firebase plugin needs java 17 and is not compatible with this plugin

**Comment by erperejildo** [comment]
same for me. I found same issue opened by me, but this time it didn't work

**Comment by theLee3** [comment]
The issue stems from building with Gradle version 8 and the new recommended Android build configurations. More accurately, the problem is that our `build.gradle` config does not include the `android.compileOptions` or `android.kotlinOptions` properties. The defaults have changed with Gradle 8, and that is causing the crash.

A patch has been created on top of games_services 4.02. To use it, please add the following to your `pubspec.yaml`:
```yaml
games_services:
  git:
      url: https://github.com/Abedalkareem/games_services
      ref: bugfix/gradle8
      path: games_services
```

**Comment by theLee3** [comment]
The issue will be officially resolved in the next release in which we will migrate to the latest Gradle & Android configurations.

**Comment by erperejildo** [comment]
> The issue will be officially resolved in the next release in which we will migrate to the latest Gradle & Android configurations thanks to the work of @AntonHolovin in Pull Request #168.

any idea when will be the next release?

**Comment by AntonHolovin** [comment]
Hi @erperejildo 

You can reference a forked repo with fixes for the time being: https://github.com/AntonHolovin/games_services.

**Comment by erperejildo** [comment]
> Hi @erperejildo 
> 
> You can reference a forked repo with fixes for the time being: https://github.com/AntonHolovin/games_services. 

Thanks. It's ok no rush. I can wait for the release

**Comment by theLee3** [comment]
I'm hoping by the end of the week. I can approve & merge the PR, but publishing will depend on when @Abedalkareem is available to do so. Shouldn't be long though.

**Comment by theLee3** [comment]
Fixed by #168
